convert_state.py

```diff
@@ -1,60 +1,111 @@
+"""
+Application to convert state names to their state codes
+and state codes to their corresponding state names
+"""
+import json 
 
 
-
-import json  # import json for json processing
 def main():
-    with open('data.json','r') as file:
-        data = file.read()
-        data = json.loads(data)
-
-    print(data)
-    
-    # Create a dictionary of states -> abbreviations, and a dictionary of abbreviations -> states 
-    stateAbbr = data #dictionary of state abbreviations keys and state name values 
-    stateAbbr2 = {}# dictionary of state name keys and state abbreviation values
-    # state key is the key in the state dictionary. state value is the value in the dictionary 
-    for statekey, statevalue in stateAbbr.items() :
-        stateAbbr2[statevalue] =  statekey
+    """ 
+    Allows the user to choose whether they'd like
+    to convert a state to state code, state code to state,
+    or quit.
+    """
+    states_file = "states.json"
 
+    # dictionary of state codes with corresponding state names
+    abrvs_to_states = read_file(states_file)
 
+    # dictionary of states with corresponding state codes
+    states_to_abrvs = reverse_dict(abrvs_to_states)  
 
-    #loop
     while True:
-        print('1. Convert state to abbreviation')
-        print("2. Convery abreviasion to state")
-        print(' 3. quit')
-        choice =   input("Enter choice : ")
 
-        if choice=="1":
+        choice = show_menu_get_choice()
+
+        if choice == "1":
+
+            convert_state_to_abvr(states_to_abrvs)
 
-            convertStateToAbbreviation(stateAbbr2)
         elif  choice == "2":
-            convert_abbreviation_to_state(stateAbbr)
-        
+
+            convert_abvr_to_state(abrvs_to_states)
         
         elif choice==  "3":
+
+            print("Good Bye.")
             break
+
         else:
-            print('try again')
-def convertStateToAbbreviation(dictionary ):
-    userInput   = input("Enter state name").  capitalize()
-    result = dictionary.get(userInput  )
-    if result == None:
-        print(' state not found')
+
+            print("Invalid entry.")
+  
+
+def convert_state_to_abvr(states):
+    """
+    Takes a state name and converts it to a state code
+
+    """
+    state = input("Enter state name: ").capitalize()
+    abrv_name = states.get(state)
```

Comment:
abbreviation or abbreviation_name would be a better name for this variable, same with this function name

Reply:
That makes sense.